src/editor/editor_brush.cpp

```diff
@@ -0,0 +1,409 @@
+{
+	switch (properties.type)
+	{
+		case EBrushTypes::SingleTile:
+			fillWith(tile, isInit ? false : true);
+			break;
+		default:
+			if (withinBounds(col, row)) {
+				tiles[col + row * width] = tile;
+			}
+	}
+}
+void  CBrush::setTiles(uint8_t srcWidth, uint8_t srcHeight, const std::vector<tile_index> &srcTiles)
+{
+	setSize(srcWidth, srcHeight);
+	fillWith(srcTiles);
+}
+
+void CBrush::fillWith(tile_index tile, bool init /* = false */)
+{
+	if (init && properties.shape == EBrushShapes::Round) {
+		ranges::fill(tiles, 0);
+		if (width == height) {
+			drawCircle(width / 2, height / 2, width, tile, this->tiles);
+		}
+	} else {
+		for (auto &brushTile : tiles) {
+			if (brushTile || init) {
+				brushTile = tile;
+			}
+		}
+	}
+	if (init) {
+		isInit = true;
+	}
+}
+
+void CBrush::fillWith(const std::vector<tile_index> &tilesSrc)
+{
+	if (tilesSrc.size() != tiles.size()) {
+		return;
+	}
+	ranges::copy(tilesSrc, tiles.begin());
+}
+
+/// Manual activated randomization of brush's tiles.
+void CBrush::randomize()
+{
+	if (!properties.randomizeAllowed) {
+		return;
+	}
+	for (auto &tile: tiles){
+		tile = randomizeTile(tile);
+	}
+}
+
+TilePos CBrush::getAllignOffset() const
+{
+	TilePos allignOffset{};
+	if (isCentered()) {
+		allignOffset.x = -getWidth() / 2;
+		allignOffset.y = -getHeight() / 2;
+	}
+	return allignOffset;
+}
+
+void CBrush::setSize(uint8_t newWidth, uint8_t newHeight)
+{
+	width = newWidth;
+	height = newHeight;
+	tiles.resize(width * height, 0);
+}
+
+void CBrush::resizeW(uint8_t newWidth)
+{
+	resize(newWidth, this->height);
+}
+
+void CBrush::resizeH(uint8_t newHeight)
+{
+	resize(this->width, newHeight);
+}
+
+void CBrush::resize(uint8_t newWidth, uint8_t newHeight)
+{
+	if (!properties.resizable) {
+		return;
+	}
+	if (newWidth != this->width && properties.resizeSteps.width == 0) {
+		return;
+	}
+	if (newHeight != this->height && properties.resizeSteps.height == 0) {
+		return;
+	}
+
+	const auto currentTile = properties.type == EBrushTypes::SingleTile ? getCurrentTile() : 0;
+
+	tiles.clear();
+	width = newWidth;
+	height = newHeight;
+
+	if(properties.shape == EBrushShapes::Round) {
+		if (newWidth && newWidth % 2 == 0) {
+			width = newWidth - 1;
+		}
+		height = width;
+	}
+	tiles.resize(width * height, 0);
+
+	if (properties.type == EBrushTypes::SingleTile) {
+		fillWith(currentTile, true);
+	}
+	/// FIXME: Init|fill
+}
+
+void CBrush::updateDecorationOption(const TDecorationOptionName &option, const TDecorationOptionValue &value)
+{
+	if (properties.decorationGenerator.options.count(option)
+		&& ranges::contains(properties.decorationGenerator.options[option], value)) {
+
+		decorationOptions[option] = value;
+		loadDecoration();
+	}
+}
+
+const CBrush::TDecorationOptionValue& CBrush::getDecorationOption(const TDecorationOptionName &option)
+{
+	static const TDecorationOptionValue emptyValue{"no value"};
+	if (!decorationOptions.count(option)) {
+		return emptyValue;
+	}
+	return decorationOptions[option];
```

Comment:
```suggestion
	if (auto it = decorationOptions.find(option); it != decorationOptions.end()) {
		return it->second;
	}
	return emptyValue;
```
would do just one look-up instead of 2

Reply:
Again, for me it's a matter of readability:
 `it->second` vs `decorationOptions[option]`

fixed:
https://github.com/Wargus/stratagus/pull/732/commits/fd85c8addb131e0e70eeade8a2a7353a793504cf